applications/vessel-history/src/features/app/app.selectors.ts

```diff
@@ -0,0 +1,18 @@
+import { createSelector } from '@reduxjs/toolkit'
+import { DEFAULT_WORKSPACE } from 'data/config'
+import { selectUrlViewport, selectUrlTimeRange } from 'routes/routes.selectors'
+
+export const selectViewport = createSelector([selectUrlViewport], (urlViewport) => {
+  return {
+    zoom: urlViewport?.zoom || DEFAULT_WORKSPACE.zoom,
+    latitude: urlViewport?.latitude || DEFAULT_WORKSPACE.latitude,
+    longitude: urlViewport?.longitude || DEFAULT_WORKSPACE.longitude,
+  }
+})
+
+export const selectTimeRange = createSelector([selectUrlTimeRange], ({ start, end }) => {
```

Comment:
Well, that's something that I left because one of the suggestions from product was that we could implement some sort of timebar to let the users navigate thru time without having to retrieve all the events at once from the api. Makes sense?